Corrfunc/mocks/DDtheta_mocks.py

```diff
@@ -351,6 +351,304 @@ def DDtheta_mocks(autocorr, nthreads, binfile,
+        import find_fastest_DDtheta_mocks_bin_refs
+    >>> binfile = pjoin(dirname(abspath(Corrfunc.__file__)),
+    ...                 "../mocks/tests/", "angular_bins")
+    >>> N = 100000
+    >>> nthreads = 4
+    >>> seed = 42
+    >>> np.random.seed(seed)
+    >>> RA1 = np.random.uniform(0.0, 2.0*pi, N)*180.0/pi
+    >>> cos_theta = np.random.uniform(-1.0, 1.0, N)
+    >>> DEC1 = 90.0 - np.arccos(cos_theta)*180.0/pi
+    >>> autocorr = 1
+    >>> best, _ = find_fastest_DDtheta_mocks_bin_refs(autocorr, nthreads,\
+    ...                                               binfile, RA1, DEC1,\
+    ...                                               return_runtimes=True)
+    >>> print(best) # doctest:+SKIP
+    (2, 1)
+
+    .. note:: Since the result might change depending on the computer,
+    doctest is skipped for this function.
+    """
+
+    weights1 = None
+    weights2 = None
+    weight_type = None
+
+    try:
+        from Corrfunc._countpairs_mocks import countpairs_theta_mocks as\
+            DDtheta_mocks_extn
+    except ImportError:
+        msg0 = "Could not import the C extension for the angular "\
+              "correlation function for mocks."
+        raise ImportError(msg0)
+
+    import numpy as np
+    from Corrfunc.utils import translate_isa_string_to_enum, fix_ra_dec,\
+        return_file_with_rbins, convert_to_native_endian, process_weights
+    from future.utils import bytes_to_native_str
+    import itertools
+    import time
+
+    weights1, weights2 = process_weights(weights1, weights2, RA1, RA2,
+                                         weight_type, autocorr)
+
+    # Ensure all input arrays are native endian
+    RA1, DEC1, weights1, RA2, DEC2, weights2 = [
+        convert_to_native_endian(arr, warn=True) for arr in
+        [RA1, DEC1, weights1, RA2, DEC2, weights2]]
+
+    fix_ra_dec(RA1, DEC1)
+    if autocorr == 0:
+        fix_ra_dec(RA2, DEC2)
+
+    # Passing None parameters breaks the parsing code, so avoid this
+    kwargs = {}
+    for k in ['weights1', 'weights2', 'weight_type', 'RA2', 'DEC2']:
+        v = locals()[k]
+        if v is not None:
+            kwargs[k] = v
+
+    integer_isa = translate_isa_string_to_enum(isa)
+    rbinfile, delete_after_use = return_file_with_rbins(binfile)
+    bin_refs = np.arange(1, maxbinref + 1)
+    if link_in_ra:
+        bin_ref_perms = itertools.product(bin_refs, bin_refs)
+    else:
+        bin_ref_perms = [(1, binref) for binref in bin_refs]
+    dtype = np.dtype([(bytes_to_native_str(b'nRA'), np.int),
+                      (bytes_to_native_str(b'nDEC'), np.int),
+                      (bytes_to_native_str(b'avg_time'), np.float),
+                      (bytes_to_native_str(b'sigma_time'), np.float)])
+    all_runtimes = np.zeros(maxbinref ** 3, dtype=dtype)
+    all_runtimes[:] = np.inf
+
+    if autocorr == 0:
+        if RA2 is None or DEC2 is None:
+            msg1 = "Must pass valid arrays for RA2/DEC2 for "\
```

Comment:
Please rename to `msg` - we pretend that this variable only exists within the enclosing `if` condition